packages/app/src/cli/services/dev/ui/components/Dev.tsx

```diff
@@ -0,0 +1,192 @@
+import {developerPreviewUpdate, disableDeveloperPreview, enableDeveloperPreview} from '../../../context.js'
+import {fetchAppFromApiKey} from '../../fetch.js'
+import {OutputProcess} from '@shopify/cli-kit/node/output'
+import {ConcurrentOutput} from '@shopify/cli-kit/node/ui/components'
+import {useAbortSignal} from '@shopify/cli-kit/node/ui/hooks'
+import React, {FunctionComponent, useEffect, useMemo, useRef, useState} from 'react'
+import {AbortController, AbortSignal} from '@shopify/cli-kit/node/abort'
+import {Box, Text, useInput, useStdin} from 'ink'
+import {handleCtrlC} from '@shopify/cli-kit/node/ui'
+import {openURL} from '@shopify/cli-kit/node/system'
+import figures from '@shopify/cli-kit/node/figures'
+import {TunnelClient} from '@shopify/cli-kit/node/plugins/tunnel'
+import {treeKill} from '@shopify/cli-kit/node/tree-kill'
+import {Writable} from 'stream'
+
+export interface DevProps {
+  processes: OutputProcess[]
+  abortController: AbortController
+  previewUrl: string
+  app: {
+    canEnablePreviewMode: boolean
+    developmentStorePreviewEnabled?: boolean
+    apiKey: string
+    token: string
+  }
+  tunnelClient?: TunnelClient
+  pollingTime?: number
+}
+
+const Dev: FunctionComponent<DevProps> = ({
+  abortController,
+  processes,
+  previewUrl,
+  app,
+  tunnelClient,
+  pollingTime = 5000,
+}) => {
+  const {apiKey, token, canEnablePreviewMode, developmentStorePreviewEnabled} = app
+  const {isRawModeSupported: canUseShortcuts} = useStdin()
+  const pollingInterval = useRef<NodeJS.Timeout>()
+  const [statusMessage, setStatusMessage] = useState(`Preview URL: ${previewUrl}`)
+
+  const {isAborted} = useAbortSignal(abortController.signal, async () => {
+    setStatusMessage('Gracefully shutting down dev ...')
```

Comment:
```suggestion
    setStatusMessage('Shutting down dev ...')
```
Don't think we need to add the "graceful" in this case. Just saying that we are shutting down implies we are doing something important for clean up I think.